### PR TITLE
Fix tslint errors and turn on tslint commit hook and on CI

### PR DESCRIPTION
**Pull request by amaranth**
The first commit in this PR fixes all reported tslint errors. The second one adds a commit hook to run tslint and try to autofix errors, like was done previously. It also changes the travis-ci configuration to run tslint and generate code coverage reports again.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,10 +16,10 @@ before_script:
   - sudo service mysql stop
   - sudo service postgresql stop
   - docker-compose up -d
-  - cp ormconfig.travis.json ormconfig.json
+  - npm run setup:config
```
Comment (pleerock) [inline .travis.yml]: this is not needed really, previous command is simple and I would prefer not to use gulp too much.

File: .travis.yml [hunk 2]
```diff
@@ -16,10 +16,10 @@ before_script:
...
 
 script:
-  - npm test
+  - npm run test-ci
```
Comment (pleerock) [inline .travis.yml]: can you please explain this change?

File: package.json [hunk 3]
```diff
@@ -91,17 +91,24 @@
...
     "*.ts": [
-      "tslint --fix",
+      "tslint --project . --fix",
       "git add"
     ]
   },
   "scripts": {
     "test-ci": "gulp ci-tests",
-    "test": "rimraf ./build && tsc && mocha --file ./build/compiled/test/utils/test-setup.js --bail --recursive --timeout 30000  ./build/compiled/test",
+    "test": "gulp tests",
```
Comment (pleerock) [inline package.json]: can you please explain this change?

**Comment by pleerock** [comment]
> Right now things are in a weird hybrid state of using gulp or just using npm. If you want to avoid gulp should I make a PR to remove gulp usage entirely? Otherwise, we should port everything back to going through gulp for consistency which this PR attempts to do.

I would like to avoid using gulp but Im not sure if its efficiently possible.

**Comment by imnotjames** [comment]
We're no longer using tslint nor travis so I'll be closing this out.  Cheers!

**Pull request closed without merging**